platform/representation/src/TemplateLinker.js

```diff
@@ -0,0 +1,140 @@
+
+define(
+    [],
+    function () {
+        "use strict";
+
+        /**
+         * The `templateLinker` service is intended for internal use by
+         * the `mct-include` and `mct-representation` directives. It is
+         * used to support common behavior of directives; specifically,
+         * loading templates and inserting them into a specified element,
+         * and/or removing that element from the DOM when there is no
+         * template to populate it with.
+         *
+         * @param $http Angular's `$http` service
+         * @param {Function} $compile Angular's `$compile` service
+         * @param $log Angular's `$log` service
+         * @private
+         */
+        function TemplateLinker($http, $compile, $log) {
+            this.templateMap = {};
+            this.$http = $http;
+            this.$compile = $compile;
+            this.$log = $log;
+        }
+
+        TemplateLinker.prototype.load = function (templateUrl) {
+            var $http = this.$http,
+                $compile = this.$compile,
+                $log = this.$log,
+                templateMap = this.templateMap;
+
+            return (templateMap[templateUrl] = templateMap[templateUrl] ||
+                $http.get(templateUrl).then(function (response) {
+                    return $compile(response.data);
+                }, function () {
+                    $log.warn("Couldn't load template at " + templateUrl);
+                    templateMap[templateUrl] = undefined;
+                }));
+        };
+
+        /**
+         * Populate the given element with templates, within the given scope;
+         * intended to support the `link` function of the supported directives.
+         *
+         * @param {Scope} scope the Angular scope to use when rendering
+         *        templates
+         * @param element the jqLite-wrapped element into which templates
+         *        should be inserted
+         * @returns {Function} a function which can be called with a template
+         *          URL to switch templates, or `undefined` to remove.
+         */
+        TemplateLinker.prototype.link = function (scope, element, templateUrl) {
+            var activeElement = element,
+                activeTemplateUrl,
+                comment = this.$compile('<!-- hidden mct element -->')(scope),
+                self = this;
+
+            function removeElement() {
+                if (activeElement !== comment) {
+                    activeElement.replaceWith(comment);
+                    activeElement = comment;
+                }
+            }
+
+            function addElement() {
+                if (activeElement !== element) {
+                    activeElement.replaceWith(element);
+                    activeElement = element;
+                    activeElement.empty();
+                }
+            }
+
+            function populateElement(template) {
+                template(scope, function (innerClone) {
```

Comment:
This is a good candidate location for this, but think this is out of scope (no pun intended) for this change. Need to make sure appropriate properties get exposed in each new scope that is created, and that scopes are appropriately synchronized with the scope of the `mct-include` or `mct-representation` itself (e.g. to pass in `parameters`), etc...